experimental/dds/tree2/src/test/shared-tree-core/branch.spec.ts

```diff
@@ -439,6 +435,54 @@ describe("Branches", () => {
 		assert.equal(branch.getHead().revision, nullRevisionTag);
 	});
 
+	describe("all nested forks and transactions are disposed and aborted when transaction is", () => {
+		const setUpNestedForks = (rootBranch: DefaultBranch) => {
+			change(rootBranch);
+			const fork1 = rootBranch.fork();
+			change(rootBranch);
+			rootBranch.startTransaction();
+			const fork2 = rootBranch.fork();
+			change(rootBranch);
+			const fork3 = rootBranch.fork();
+			change(fork3);
```

Comment:
This test does not check that nested forks are disposed even if their parent branch isn't in the middle of a transaction.
```suggestion
			const fork3 = rootBranch.fork();
			const testMe = fork3.fork();
			change(fork3);
```
`testMe` should end up disposed, which I don't think your current implementation handles properly.